java-api/src/main/java/com/db/grad/javaapi/service/DogHandler.java

```diff
@@ -17,4 +19,9 @@ public Long addDog(Dog theDog) {
     public long getNoOfDogs() {
         return itsDogRepo.count();
     }
+
+    public boolean removeDog(long id){
+        return itsDogRepo.delete(itsDogRepo.findById(id));
```

Comment:
What if there is no dog with such an ID? U sure that it will not throw when you pass null to delete? 
At least please add a test with an incorrect id.